src/Groups/group_constructors.jl

```diff
@@ -0,0 +1,323 @@
+  end
+  return PermGroup(GAP.Globals.AlternatingGroup(n), n)
+end
+
+function alternating_group(::Type{T}, n::Int) where T <: Group
+  if n < 1
+    throw(ArgumentError("it must be a positive integer"))
+  end
+  return T(GAP.Globals.AlternatingGroup(_gap_filter(T), n))
+end
+
+function isalternating_group(G::Group)
+  return GAP.Globals.IsAlternatingGroup(G.X)
+end
+
+function small_group(n::Int, m::Int)
+  G = GAP.Globals.SmallGroup(n, m)
+  T = _get_type(G)
+  return T(G)
+end
+
+function small_groups_id(G::Group)
+  r = GAP.Globals.IdGroup(G.X)
+  res = GAP.gap_to_julia(GAP.gap_to_julia(r))
+  return (res[1], res[2])
+end
+
+function transitive_group(n::Int, m::Int)
+  return PermGroup(GAP.Globals.TransitiveGroup(n, m))
+end
+
+function cyclic_group(n::Int)
+  return PcGroup(GAP.Globals.CyclicGroup(n))
+end
+
+function cyclic_group(::Type{T}, n::Int) where T <: Group
+  return T(GAP.Globals.CyclicGroup(_gap_filter(T), n))
+end
+
+function iscyclic(G::Group)
+  return GAP.Globals.IsCyclic(G.X)
+end
+
+# already defined in Hecke
+#=
+function abelian_group(v::Vector{Int})
+  for i = 1:length(v)
+    iszero(v[i]) && error("Cannot represent an infinite group as a polycyclic group")
+  end
+  v1 = GAP.julia_to_gap(v)
+  return PcGroup(GAP.Globals.AbelianGroup(v1))
+end
+=#
+
+"""
+    abelian_group(::Type{T}, v::Vector{Int}) where T <: Group -> PcGroup
+return the direct product of cyclic groups of order v[1] x v[2] x ... x v[n], as group of type `T`. Here, `T` must be of type `PermGroup`, `FPGroup` or `PcGroup`.
+"""
+function abelian_group(::Type{T}, v::Vector{Int}) where T <: Group
+  v1 = GAP.julia_to_gap(v)
+  return T(GAP.Globals.AbelianGroup(_gap_filter(T), v1))
+end
+
+"""
+    isabelian(G::Group)
+returns whether `G` is abelian.
+"""
+function isabelian(G::Group)
+  return GAP.Globals.IsAbelian(G.X)
+end
+
+function mathieu_group(n::Int)
+  @assert n in Int[9, 10, 11, 12, 21, 22, 23, 24]
+  return PermGroup(GAP.Globals.MathieuGroup(n), n)
+end
+
+
+################################################################################
+#
+# begin FpGroups
+#
+################################################################################
+
+"""
+    free_group
+There are three ways to define a free group.
+- `free_group(n::Int) -> FPGroup`; return the free group of rank `n`, with generators printed as `"f1"`,`"f2"`,`"f3"`, etc.
+- `free_group(L::String...) -> FPGroup`; return the free group with length(`L`) generators, printed as `L[1]`, `L[2]`, `L[3]`, etc.
+- `free_group(n::Int, s::String) -> FPGroup`; return the free group of rank `n`, with generators printed as `"s1"`, `"s2"`, `"s3"`, etc.
+
+!!! warning "Note"
+    In every case, it is *not* defined a variable named as the generators are printed.
+"""
+function free_group(n::Int)
+   return FPGroup(GAP.Globals.FreeGroup(n))
+end
+
+function free_group(L::String...)
+   J=GAP.julia_to_gap([GAP.julia_to_gap(x) for x in L])
+   return FPGroup(GAP.Globals.FreeGroup(J))
+end
```

Comment:
@GDeFranceschi there should be another variant here (with test, and documented above): one which takes an `Array{String,1}`